src/index.tsx

```diff
@@ -1,16 +1,55 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import {
+  createBrowserRouter,
+  RouterProvider,
+  redirect
+} from "react-router-dom";
 import "./index.css";
 import App from "./App";
 import logo from "./logo.svg";
 
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <App />,
+    loader: ({ params }) => {
+      // If the app was accessed through old URL format, redirect
+      const txhash = new URLSearchParams(window.location.search).get("t");
+
+      // This loader is executed when accessing tx/:txhash, we need this check to avoid infinite loop
+      if (txhash?.length === 66 && !params.txhash) {
+        return redirect(`tx/${txhash}`);
+      }
+    },
+    children: [
+      {
+        path: "tx",
+        element: null,
+        loader: ({ params }) => {
+          const { txhash } = params;
+          if (!txhash || txhash.length !== 66) {
+            return redirect("/");
```

Comment:
We might as well add some very light validation, e.g. https://ethereum.stackexchange.com/a/72625

Reply:
Added a function for it